adapters/incluster/v1/client.go

```diff
@@ -415,3 +434,122 @@ func (c *Client) getObjectFromUnstructured(d *unstructured.Unstructured) ([]byte
 	}
 	return utils.FilterAndMarshal(d)
 }
+
+// Batch processing functions
+func reconcileBatchProcessingFunc(ctx context.Context, c *Client, items domain.BatchItems) error {
+	logger.L().Debug("reconciliation batch started", helpers.String("resource", c.res.Resource))
+	if len(items.NewChecksum) == 0 {
+		return fmt.Errorf("reconciliation batch (%s) was empty - expected at least one NewChecksum message", c.res.Resource)
+	}
+
+	list, err := c.client.Resource(c.res).Namespace("").List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("list resources: %w", err)
+	}
+
+	for _, item := range items.NewChecksum {
+		idx := findResourceInList(list.Items, item.Namespace, item.Name)
+
+		id := domain.KindName{
+			Kind:            c.kind,
+			Name:            item.Name,
+			Namespace:       item.Namespace,
+			ResourceVersion: item.ResourceVersion,
+		}
+
+		// resource was not found, sending delete message
+		if idx == -1 {
+			logger.L().Debug("resource missing", helpers.String("kind", item.Kind.String()), helpers.String("name", item.Name), helpers.String("namespace", item.Namespace))
+			err = multierr.Append(err, c.callbacks.DeleteObject(ctx, id))
+			continue
+		}
+
+		// resource was found
+		resource := list.Items[idx]
+		currentVersion := domain.ToResourceVersion(resource.GetResourceVersion())
+		if currentVersion == item.ResourceVersion {
+			// resource has same version, skipping
+			logger.L().Debug("resource has same version", helpers.String("resource", item.Kind.String()), helpers.String("name", item.Name), helpers.String("namespace", item.Namespace))
+			continue
+		}
+
+		// resource has changed, sending a put message
+		logger.L().Debug("resource has changed",
+			helpers.String("resource", item.Kind.String()),
+			helpers.String("name", item.Name),
+			helpers.String("namespace", item.Namespace),
+			helpers.Int("batch resource version", item.ResourceVersion),
+			helpers.Int("current resource version", currentVersion))
+		newObject, marshalErr := c.getObjectFromUnstructured(&resource)
+		if marshalErr != nil {
+			err = multierr.Append(err, fmt.Errorf("marshal resource: %w", marshalErr))
+		}
+		err = multierr.Append(err, c.callbacks.PutObject(ctx, id, newObject))
+	}
+
+	return nil
+}
+
+func findResourceInList(list []unstructured.Unstructured, namespace, name string) int {
```

Comment:
TODO for the future use a map once we hit bottlenecks because our customers have 8k objects (I think other things will limit us first)

Reply:
You suggest to build a map and then use it for lookups or have the k8s API return a map somehow?